keras_cv/models/segmentation/segment_anything/sam_prompt_encoder.py

```diff
@@ -0,0 +1,321 @@
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import math
+
+from keras_cv.api_export import keras_cv_export
+from keras_cv.backend import keras
+from keras_cv.backend import ops
+from keras_cv.layers.serializable_sequential import SerializableSequential
+
+
+@keras_cv_export("keras_cv.layers.RandomFrequencyPositionalEmbeddings")
+class RandomFrequencyPositionalEmbeddings(keras.layers.Layer):
+    def __init__(self, num_positional_features, scale, **kwargs):
+        """Positional encoding using random spatial frequencies.
+
+        This layer maps coordinates/points in 2D space to positional
+        encodings using random spatial frequencies.
+
+        Args:
+            num_positional_features (int): Number of positional features
+                in the output.
+            scale (float): The standard deviation of the random frequencies.
+        """
+        super().__init__(**kwargs)
+        self.num_positional_features = num_positional_features
+        self.scale = scale
+        init_func = lambda *a, **kw: self.scale * ops.random.normal(
+            shape=(2, self.num_positional_features), dtype=self.dtype
+        )
+        self.positional_encoding_gaussian_matrix = self.add_weight(
+            name="positional_emcoding_gaussian_matrix",
+            shape=(2, self.num_positional_features),
+            dtype=self.dtype,
+            trainable=False,
+            initializer=init_func,
+        )
+
+        self.built = True
+
+    def __positional_encodings(self, coords):
+        coords = 2 * coords - 1
+        coords = coords @ self.positional_encoding_gaussian_matrix
+        coords = 2 * math.pi * coords
+        return ops.concatenate([ops.sin(coords), ops.cos(coords)], axis=-1)
+
+    def call(self, size):
+        """Generate a positional encoding for an image of any given size.
+
+        Args:
+            size (int): The size of the image.
+
+        Returns:
+            tensor: Positional encoding of the image.
+        """
+        H, W = size
+        H, W = ops.cast(H, "int64"), ops.cast(W, "int64")
+        grid = ops.ones(shape=(H, W), dtype=self.dtype)
+        y_embed = ops.cumsum(grid, axis=0) - 0.5
+        x_embed = ops.cumsum(grid, axis=1) - 0.5
+        y_embed = y_embed / ops.cast(H, self.dtype)
+        x_embed = x_embed / ops.cast(W, self.dtype)
+        return self.__positional_encodings(
+            ops.stack([x_embed, y_embed], axis=-1)
+        )
+
+    def call_with_coords(self, coords_input, image_size):
```

Comment:
Maybe refactoring to a cleaner name? "encode_coordinates" sounds a lot nicer than the official "call_with_coords" IMO

Reply:
Done.